### PR TITLE
vendor: check in vendor trees

### PR DESCRIPTION
**Pull request by dongsupark**
To be able to run smoke tests remotely on SemaphoreCI or else, we need to optimize build performance for each build. For that, we need to get the vendor tree included in the source tree. Otherwise running `dep ensure` for each build could bring negative impact on build performance.

NOTE: this PR might include unnecessary files, such as test files or documents. However, at the moment `dep` doesn't provide a native way of excluding such files. Thus I made them included for now.

Fixes https://github.com/kinvolk/kube-spawn/issues/153

### CODE REVIEW

**Comment by iaguis** [comment]
It doesn't seem to compile for me:

```
$ make
go build -o cni-noop ./cmd/cni-noop
go build -o cnispawn ./cmd/cnispawn
go build -o kube-spawn-runc ./cmd/kube-spawn-runc
go build -o kube-spawn \
	-ldflags "-X main.version=v0.1.1-34-g7240c2b" \
	./cmd/kube-spawn
# github.com/kinvolk/kube-spawn/vendor/github.com/docker/docker/api/types
vendor/github.com/docker/docker/api/types/types.go:47:6: GraphDriverData redeclared in this block
	previous declaration at vendor/github.com/docker/docker/api/types/graph_driver_data.go:8:6
# github.com/kinvolk/kube-spawn/vendor/github.com/docker/distribution/reference
vendor/github.com/docker/distribution/reference/normalize.go:34:11: undefined: anchoredIdentifierRegexp
vendor/github.com/docker/distribution/reference/normalize.go:84:28: undefined: namedRepository
vendor/github.com/docker/distribution/reference/normalize.go:85:10: invalid type for composite literal: repository
vendor/github.com/docker/distribution/reference/normalize.go:102:18: unknown field 'namedRepository' in struct literal of type reference
vendor/github.com/docker/distribution/reference/normalize.go:114:18: unknown field 'namedRepository' in struct literal of type taggedReference
vendor/github.com/docker/distribution/reference/normalize.go:121:18: unknown field 'namedRepository' in struct literal of type canonicalReference
vendor/github.com/docker/distribution/reference/normalize.go:145:11: undefined: anchoredIdentifierRegexp
vendor/github.com/docker/distribution/reference/normalize.go:158:11: undefined: anchoredShortIdentifierRegexp
make: *** [Makefile:9: all] Error 2
$ make vendor all
/home/iaguis/work/go/bin/dep
dep ensure
ensure Solve(): No versions of golang.org/x/sys met constraints:
	master: Unable to update checked out version: fatal: reference is not a tree: 314a259e304ff91bd6985da2a7149bbf91237993

	master: Unable to update checked out version: fatal: reference is not a tree: 314a259e304ff91bd6985da2a7149bbf91237993

make: *** [Makefile:14: vendor] Error 1
```



Also, when it works, should we change documentation to `make all` or `make` instead of `make vendor all`?

We should probably setup at least some simple CI that builds the project inside a docker container so we're sure the vendoring is done right.

**Comment by dongsupark** [comment]
@iaguis Build fails probably because the previous uncommitted vendor tree remains. Can you please remove the vendor completely, and check it out again, like `rm -rf vendor; git checkout vendor`?
And yes, we should change the document to `make all` instead of `make vendor all`.
I'll try to set up travis CI as well.

**Comment by iaguis** [comment]
> @iaguis Build fails probably because the previous uncommitted vendor tree remains.

You're right. Thanks!

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,31 @@
...
 all:
 	go build -o cni-noop ./cmd/cni-noop
 	go build -o cnispawn ./cmd/cnispawn
 	go build -o kube-spawn-runc ./cmd/kube-spawn-runc
 	go build -o kube-spawn \
 		-ldflags "-X main.version=$(VERSION)" \
 		./cmd/kube-spawn
+endif
 
-vendor: | dep
```
Comment (iaguis) [inline Makefile]: Should we keep this and call it `update-vendor` or similar?

Comment (dongsupark) [reply Makefile]: @iaguis Done

**Pull request merged**